### ISSUE TITLE
Issue with 2.0

### ISSUE THREAD

**Issue by milancermak** [matt-filion/serverless-external-s3-event#19]
I tried using the version that's currently in 2.0, because 1.0.6 didn't work for me when trying to resolve bugs mentioned in #12, #13, #14, #16 

However, when I added `serverless-plugin-existing-s3` to the plugins section of my serverless.yml, I got an error when trying to deploy.

```
$ SLS_DEBUG=* sls deploy
Serverless: WARNING: Plugin ExistingEventRulePlugin uses deprecated hook deploy:compileEvents

  Syntax Error -------------------------------------------

     Unexpected string

     For debugging logs, run again after setting the "SLS_DEBUG=*" environment variable.

  Stack Trace --------------------------------------------

SyntaxError: Unexpected string
    at Object.exports.runInThisContext (vm.js:78:16)
    at Module._compile (module.js:543:28)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at plugins.forEach (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:98:22)
    at Array.forEach (native)
    at PluginManager.loadPlugins (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:97:13)
    at PluginManager.loadServicePlugins (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:124:10)
    at PluginManager.loadAllPlugins (/usr/local/lib/node_modules/serverless/lib/classes/PluginManager.js:93:10)
    at service.load.then (/usr/local/lib/node_modules/serverless/lib/Serverless.js:64:28)
From previous event:
    at runCallback (timers.js:651:20)
    at tryOnImmediate (timers.js:624:5)
    at processImmediate [as _immediateCallback] (timers.js:596:5)
From previous event:
    at __dirname (/usr/local/lib/node_modules/serverless/bin/serverless:15:28)
    at Object.<anonymous> (/usr/local/lib/node_modules/serverless/bin/serverless:27:4)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.runMain (module.js:605:10)
    at run (bootstrap_node.js:420:7)
    at startup (bootstrap_node.js:139:9)
    at bootstrap_node.js:535:3

  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Forums:        forum.serverless.com
     Chat:          gitter.im/serverless/serverless

  Your Environment Information -----------------------------
     OS:                 darwin
     Node Version:       7.4.0
     Serverless Version: 1.12.1
```

Let me know if I can help with debugging this.

**Comment by matt-filion** [comment]
So you have any other plugins in use? This plugin doesn't use the deploy
command yet, or the compileEvents lifecycle hook. Assuming you have nothing
else in the file, If you could send me your server less file simplified
down to just the pertinent function, that will help.

**Comment by milancermak** [comment]
Yes, I had one other plugin enabled, the [serverless-plugin-existing-cloudwatch-rule](https://github.com/AlexanderMS/serverless-plugin-existing-cloudwatch-rule). That's where the warning comes from, but not the error.

I don't have the exact config anymore, as I had to deploy and connect the S3 notifications to my Lambda function with other means, but the setup was very basic, something along the lines of:

```yaml
service: reporter

plugins:
  - serverless-plugin-existing-s3

provider:
  name: aws
  runtime: python3.6

functions:
  loadReports:
    handler: load_reports.handler
    events:
      - existingS3:
          bucket: bucket-outside-of-stack
          events:
            - s3:ObjectCreated:*
          rules:
            - prefix: source/
            - suffix: .csv
```

I installed the plugin via `npm install 'git://github.com/matt-filion/serverless-external-s3-event.git#2.0' --save`

**Comment by alejobs** [comment]
Hi,

I have just updated the plugin to 2.0 and I am also facing with the same error. I have no other plugins configured for this serverless project.

Thanks.

**Comment by matt-filion** [comment]
I've done another update to 2.0, this should work. Some sloppy stuff got committed, need to do a bit of research to see if there is anything else that might be missing but wont be able to do that until Saturday.

**Comment by alejobs** [comment]
Hi @matt-filion,

I have updated the plugin and now I get this error:

`Syntax Error -------------------------------------------

   Unexpected token this

   For debugging logs, run again after setting the "SLS_DEBUG=*" environment variable.

Get Support --------------------------------------------
   Docs:          docs.serverless.com
   Bugs:          github.com/serverless/serverless/issues
   Forums:        forum.serverless.com
   Chat:          gitter.im/serverless/serverless

Your Environment Information -----------------------------
   OS:                 win32
   Node Version:       6.9.2
   Serverless Version: 1.13.0`

**Comment by attinderseesdragons** [comment]
i am getting same error as @alejobs

**Comment by matt-filion** [comment]
Sorry for the inappropriately long delay on this. I think i got it fixed with 2.0.2. Its pushed to NPM as well.

**Comment by alejobs** [comment]
Hi @matt-filion,

I have tried latest version (2.0.2) and now it works. However, I have detected another bug: it removes any other S3 notification already present on the same bucket...
